request_test.go

```diff
@@ -365,3 +365,18 @@ func TestHandleHTTPHEADRequest(t *testing.T) {
 	interceptor.HandleHTTP(w, req, handler)
 	st.Expect(t, string(w.Body), "Hi")
 }
+
+func TestHandleHTTPOPTIONSRequest(t *testing.T) {
+	modifierFunc := func(m *RequestModifier) {
+		m.Header.Set("foo", "bar")
+	}
+	w := utils.NewWriterStub()
+	req := &http.Request{Method: "OPTIONS", Header: make(http.Header)}
+	handler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		fmt.Fprint(w, "Hi")
+		st.Expect(t, r.Header.Get("foo"), "")
```

Comment:
No no, it's fine as it is. This is the expected behavior since Go it's strict in terms of type enforcement.

I was pointing that out just because I think the test is not working, since the request should be intercepted and the header should be added, but seems like it doesn't. In other words, seems like `modifierFunc` is not called.
Maybe I'm wrong because it's a bit late and my brain could not work very well :P